src/neon/oapv_sad_neon.c

```diff
@@ -34,140 +34,56 @@
 
 #if ARM_NEON
 
+#if defined(__aarch64__)
+
+#define VADDVQ_S32_(s, v) \
+    s += vaddvq_s32(v);
+#define VADDVQ_S64_(s, v) \
+    s += vaddvq_s64(v);
+
+#else // __aarch64__
+
+#define VADDVQ_S32_(s, v) \
+    { \
+        int32x2_t tmp = vadd_s32(vget_low_s32(v), vget_high_s32(v)); \
+        tmp = vpadd_s32(tmp, tmp); \
+        s += vget_lane_s32(tmp, 0); \
+    }
+#define VADDVQ_S64_(s, v) \
+    s += vgetq_lane_s64(v, 0) + vgetq_lane_s64(v, 1);
+#define vabal_high_s16(a, b, c) \
+    vabal_s16(a, vget_high_s16(b), vget_high_s16(c))
+#define vsubl_high_s16(a, b) \
+    vsubl_s16(vget_high_s16(a), vget_high_s16(b))
+#define vmlal_high_s32(a, b, c) \
+    vmlal_s32(a, vget_high_s32(b), vget_high_s32(c))
+
+#endif // __aarch64__
+
 /* SAD for 16bit **************************************************************/
-int sad_16b_neon_8x2n(int w, int h, void *src1, void *src2, int s_src1, int s_src2)
+static int sad_16b_neon_8x2n(int w, int h, void *src1, void *src2, int s_src1, int s_src2)
 {
     int sad = 0;
     s16* s1 = (s16*) src1;
     s16* s2 = (s16*) src2;
     int16x8_t s1_vector, s2_vector;
-    int32x4_t  diff_part1, diff_part2, diff_part1_abs, diff_part2_abs, sad_vector, sad_vector_temp;
-    // Loop unrolled    
-    { // Row 0
+    int32x4_t  sad_vector = vdupq_n_s32(0);
+    // Loop unrolling
+#pragma GCC unroll 8
```

Comment:
Can this work on non-GCC compiler?

Reply:
It means that the unrolling will not work properly on MSVC, and it can lead slow operation.

Reply:
It's hard to tell without measurement. Measure before Optimizing.
1) Small cycle will not flush cache so it may works faster without unrolling.
2) MSVC may unroll it without pragmas. Believe to compiler.
3) Do you really use MSVC to compile ARM project?